ddtrace/contrib/celery/task.py

```diff
@@ -0,0 +1,31 @@
+from .app import patch_app
+
+from ...utils.deprecation import deprecation
+
+
+def patch_task(task, pin=None):
+    """Deprecated API. The new API uses signals that can be activated via
+    patch(celery=True) or through `ddtrace-run` script. Using this API
+    enables instrumentation on all tasks.
+    """
+    deprecation(
+        name='ddtrace.contrib.celery.patch_task',
+        message='Use `patch(celery=True)` or `ddtrace-run` script instead',
+        version='1.0.0',
+    )
+
+    # Enable instrumentation everywhere
+    patch_app(task.app)
```

Comment:
I'm not sure if this behavior is correct. Probably if developers want to instrument only 2 or 3 tasks, it's better to make this a no-op too. Otherwise they will see everything instrumented. I think it's a matter of tradeoffs here.

Reply:
Yeah, this one feels pretty rough.

My initial reaction is to say that this should be a no-op as normally that's safer than having a more expansive execution than the original semantics.

That being said, after chewing on it a bit, I think this (patch everything) is the right call. The reason being, for us `too much data` > `lost data`. If somebody were to apply an update without noticing the deprecation and we made this a no-op, they'd irretrievably loose data. On the other hand, if their account gets spammed with more than they wanted it would both be obvious where the problem is and they'd still have access to the traces they cared about (among the noise).

Reply:
OK, I will keep it that way. My reasoning to not provide a `patched_tasks` config, is because I would like to understand if it's really a use case. Also, we don't have configurations in place right now, so it's hard to provide this setting for Celery. I may work on it in another PR.